loaders/src/glTF/2.0/glTFLoader.ts

```diff
@@ -777,7 +781,7 @@ export class GLTFLoader implements IGLTFLoader {
 
         this.logOpen(`${context}`);
 
-        const shouldInstance = this._parent.createInstances && (node.skin == undefined && !mesh.primitives[0].targets);
+        const shouldInstance = this._parent.createInstances && (node.skin == undefined && !mesh.primitives[0].targets) && (!primitive._instanceData || !primitive._instanceData.babylonSourceMesh.hasThinInstances);
```

Comment:
I don't understand why we need additional conditions here? This check is for instanced meshes which have nothing to do with thin instances? Perhaps I'm missing something.

Reply:
See above:
> The problem comes from the fact that as an optimization, the glTF loader creates an InstancedMesh instead of a Mesh when possible (see line 784 in glTFLoader). But if this mesh has "gpu instances" (as per the extension), creating thin instances for it won't work because InstancedMesh don't support thin instances.

In the **teapots_galore.glb** file you have two nodes, each using the "gpu instancing" extension. The two nodes point to the same mesh definition. So, when processing the second node, the code you pointed to create an `InstancedMesh` instead of a `Mesh` and so thin instances can't be used anymore for this mesh.